### PR TITLE
feat: switch Rust startup to Rust only

### LINKED ISSUE #1243: Load config in Rust
The config crate allows TOML/env reading so that we should be able to load config in Rust directly and avoid Python for starting up the connection node.

### PR DESCRIPTION
**Pull request by bbangert**
Co-authored by: Phil Jenvey <pjenvey@underboss.org>

Closes #1243

### CODE REVIEW

**Comment by alexcrichton** [comment]
🎉

File: autopush_rs/src/settings.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            self.router_port
+        )
+    }
+
+    pub fn endpoint_url(&self) -> String {
+        let hostname = self.host_name();
+        format!(
+            "{}://{}:{}",
+            self.endpoint_scheme,
+            self.endpoint_hostname.as_ref().unwrap_or(&hostname),
```
Comment (pjenvey) [inline autopush_rs/src/settings.rs]: for just the connection node it doesn't make sense to fallback to hostname

File: Dockerfile.python27 [hunk 2]
```diff
@@ -14,7 +14,7 @@ RUN \
     make clean && \
     pip install -r requirements.txt && \
     python setup.py develop && \
-    cd autopush_rs && \
+    cargo install && \
```
Comment (pjenvey) [inline Dockerfile.python27]: needs --path

File: autopush_rs/src/settings.rs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    pub router_ssl_cert: Option<String>,
+    pub router_ssl_dh_param: Option<String>,
+    pub auto_ping_interval: f64,
+    pub auto_ping_timeout: f64,
+    pub max_connections: u32,
+    pub close_handshake_timeout: u32,
+    pub endpoint_scheme: String,
+    pub endpoint_hostname: Option<String>,
+    pub endpoint_port: u16,
+    pub crypto_key: String,
```
Comment (pjenvey) [inline autopush_rs/src/settings.rs]: this is an append (List[str]) in main_argparse. Can Config do something similar?

Comment (bbangert) [reply autopush_rs/src/settings.rs]: Not if we want it to work with env vars.

File: autopush_rs/src/settings.rs (hunk 1 above)
Comment (bbangert) [reply autopush_rs/src/settings.rs]: It's used to generate the endpoint, it needs some value from somewhere.

Comment (pjenvey) [reply autopush_rs/src/settings.rs]: Can we just start expecting it here then? The python's handling of hostnames is pretty confusing

Comment (bbangert) [reply autopush_rs/src/settings.rs]: The tests assume the endpoint hostname will be set to the hostname. They'd all have to change too.

**Pull request merged**